--- a/chrome/browser/ui/window_feature_controller/window_feature_controller.cc
+++ b/chrome/browser/ui/window_feature_controller/window_feature_controller.cc
@@ -56,17 +56,11 @@ bool WindowFeatureController::CanSupport
 
 #if BUILDFLAG(IS_MAC)
 bool WindowFeatureController::UsesImmersiveFullscreenMode() const {
-  const bool is_pwa = app_controller_ != nullptr;
-  const bool is_tabbed_window =
-      CanSupportWindowFeature(WindowFeature::kFeatureTabStrip);
-  return is_pwa || is_tabbed_window;
+  return false;
 }
 
 bool WindowFeatureController::UsesImmersiveFullscreenTabbedMode() const {
-  const bool is_pwa = app_controller_ != nullptr;
-  const bool is_tabbed_window =
-      CanSupportWindowFeature(WindowFeature::kFeatureTabStrip);
-  return is_tabbed_window && !is_pwa;
+  return false;
 }
 #endif
 
