--- a/gpu/config/gpu_finch_features.cc
+++ b/gpu/config/gpu_finch_features.cc
@@ -53,7 +53,7 @@
 // More aggressive behavior for the shader cache: increase size, and do not
 // purge as much in case of memory pressure.
-BASE_FEATURE(kAggressiveShaderCacheLimits, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kAggressiveShaderCacheLimits, base::FEATURE_ENABLED_BY_DEFAULT);

 #if BUILDFLAG(IS_ANDROID)
 // Use android SurfaceControl API for managing display compositor's buffer queue
@@ -139,7 +139,7 @@
 #if BUILDFLAG(IS_MAC) || BUILDFLAG(IS_IOS)
 // If enabled, the TASK_CATEGORY_POLICY value of the GPU process will be
 // adjusted to match the one from the browser process every time it changes.
-BASE_FEATURE(kAdjustGpuProcessPriority, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kAdjustGpuProcessPriority, base::FEATURE_ENABLED_BY_DEFAULT);
 #endif

 // When enabled, Grshader disk cache will be cleared on startup if any cache
@@ -320,7 +320,7 @@
 // be overridden by
 // --enable-skia-graphite-precompilation and
 // --disable-skia-graphite-precompilation.
-BASE_FEATURE(kSkiaGraphitePrecompilation, base::FEATURE_DISABLED_BY_DEFAULT);
+BASE_FEATURE(kSkiaGraphitePrecompilation, base::FEATURE_ENABLED_BY_DEFAULT);

 BASE_FEATURE(kConditionallySkipGpuChannelFlush,
