--- a/chrome/browser/ui/views/toolbar/toolbar_view.cc
+++ b/chrome/browser/ui/views/toolbar/toolbar_view.cc
@@ -383,15 +383,12 @@ void ToolbarView::Init() {
   PrefService* const prefs = browser_->profile()->GetPrefs();
 
   std::unique_ptr<ExtensionsToolbarDesktop> extensions_container;
-  std::unique_ptr<ToolbarDivider> toolbar_divider;
 
   // Do not create the extensions or browser actions container if it is a guest
   // profile (only regular and incognito profiles host extensions).
   if (!browser_->profile()->IsGuestSession() &&
       !features::IsWebUIExtensionsContainerEnabled()) {
     extensions_container = std::make_unique<ExtensionsToolbarDesktop>(browser_);
-
-    toolbar_divider = std::make_unique<ToolbarDivider>();
   }
 
   std::unique_ptr<MediaToolbarButtonView> media_button;
@@ -490,10 +487,6 @@ void ToolbarView::Init() {
                                                        extensions_container_);
   }
 
-  if (toolbar_divider) {
-    toolbar_divider_ = AddChildView(std::move(toolbar_divider));
-  }
-
   if (!features::IsWebUIPinnedToolbarActionsEnabled()) {
     pinned_toolbar_actions_container_ = AddChildView(
         std::make_unique<PinnedToolbarActionsContainer>(browser_view_, this));
@@ -1581,13 +1574,6 @@ void ToolbarView::InitLayout() {
             .WithOrder(kToolbarActionsFlexOrder));
   }
 
-  if (toolbar_divider_) {
-    toolbar_divider_->SetProperty(
-        views::kMarginsKey,
-        gfx::Insets::VH(
-            0, GetLayoutConstant(LayoutConstant::kToolbarDividerSpacing)));
-  }
-
    if (avatar_ &&
       base::FeatureList::IsEnabled(features::kToolbarProfileChipResizing)) {
     // Flex order for the profile avatar button is determined by the
@@ -1678,11 +1664,6 @@ void ToolbarView::LayoutCommon() {
   }
   app_menu_button_->SetTrailingMargin(
       extend_buttons_to_edge ? interior_margin.right() : 0);
-
-  if (toolbar_divider_ && extensions_container_) {
-    views::ManualLayoutUtil(layout_manager_)
-        .SetViewHidden(toolbar_divider_, !extensions_container_->GetVisible());
-  }
   // Cast button visibility is controlled externally.
 }
 
--- a/chrome/browser/ui/views/toolbar/pinned_toolbar_actions_container.cc
+++ b/chrome/browser/ui/views/toolbar/pinned_toolbar_actions_container.cc
@@ -167,6 +167,7 @@ PinnedToolbarActionsContainer::PinnedToo
               GetLayoutConstant(LayoutConstant::kToolbarIconDefaultMargin),
           0, GetLayoutConstant(LayoutConstant::kToolbarDividerSpacing)));
   toolbar_divider_ = AddChildView(std::move(toolbar_divider));
+  toolbar_divider_->SetVisible(false);
 
   // Initialize the pinned action buttons.
   action_view_controller_ = std::make_unique<views::ActionViewController>();
