--- a/chrome/browser/ui/toasts/toast_view.cc
+++ b/chrome/browser/ui/toasts/toast_view.cc
@@ -5,24 +5,24 @@
 #include "chrome/browser/ui/toasts/toast_view.h"
 
 #include <algorithm>
-#include <climits>
 #include <memory>
 #include <utility>
 
 #include "base/functional/bind.h"
 #include "chrome/app/vector_icons/vector_icons.h"
 #include "chrome/browser/ui/browser_element_identifiers.h"
+#include "chrome/browser/ui/color/chrome_color_id.h"
 #include "chrome/browser/ui/views/chrome_layout_provider.h"
 #include "chrome/browser/ui/views/chrome_typography.h"
 #include "chrome/grit/generated_resources.h"
 #include "components/strings/grit/components_strings.h"
 #include "components/vector_icons/vector_icons.h"
-#include "ui/base/accelerators/accelerator.h"
 #include "ui/base/interaction/element_identifier.h"
 #include "ui/base/l10n/l10n_util.h"
 #include "ui/base/metadata/metadata_impl_macros.h"
 #include "ui/base/models/menu_model.h"
 #include "ui/base/mojom/menu_source_type.mojom.h"
+#include "ui/color/color_id.h"
 #include "ui/compositor/layer.h"
 #include "ui/views/accessibility/view_accessibility.h"
 #include "ui/views/animation/animation_builder.h"
@@ -45,14 +45,16 @@
 namespace {
 constexpr int kAnimationEntryDuration = 300;
 constexpr int kAnimationExitDuration = 150;
-constexpr int kAnimationHeightOffset = 50;
 constexpr float kAnimationHeightScale = 0.5;
+constexpr int kBubbleCornerRadius = 8;
+constexpr int kBubbleStrokeWidth = 1;
+constexpr ui::ColorId kToastForegroundColorId =
+    kColorTabForegroundActiveFrameActive;
 
 gfx::Transform GetScaleTransformation(gfx::Rect bounds) {
   gfx::Transform transform;
-  transform.Translate(0,
-                      bounds.CenterPoint().y() * (1 - kAnimationHeightScale));
-  transform.Scale(1, kAnimationHeightScale);
+  transform.Translate(bounds.CenterPoint().x(), 0);
+  transform.Scale(kAnimationHeightScale, kAnimationHeightScale);
   return transform;
 }
 
@@ -109,18 +111,18 @@ ToastView::ToastView(
     std::optional<ui::ImageModel> image_override,
     bool render_toast_over_web_contents,
     base::RepeatingCallback<void(ToastCloseReason)> toast_close_callback)
-    : BubbleDialogDelegateView(anchor_view, views::BubbleBorder::NONE),
-      AnimationDelegateViews(this),
+    : BubbleDialogDelegateView(anchor_view,
+                               views::BubbleBorder::NONE,
+                               views::BubbleBorder::NO_SHADOW),
       toast_text_(toast_text),
       icon_(icon),
       image_override_(std::move(image_override)),
       render_toast_over_web_contents_(render_toast_over_web_contents),
       toast_close_callback_(std::move(toast_close_callback)) {
-  SetBackgroundColor(ui::kColorToastBackgroundProminent);
+  SetBackgroundColor(ui::kColorSysOmniboxContainer);
   SetShowCloseButton(false);
   DialogDelegate::SetButtons(static_cast<int>(ui::mojom::DialogButton::kNone));
-  set_corner_radius(ChromeLayoutProvider::Get()->GetDistanceMetric(
-      DISTANCE_TOAST_BUBBLE_HEIGHT));
+  set_corner_radius(kBubbleCornerRadius);
   SetProperty(views::kElementIdentifierKey, kToastElementId);
   set_close_on_deactivate(false);
   SetProperty(views::kElementIdentifierKey, kToastViewId);
@@ -161,6 +163,14 @@ int ToastView::GetIconSize() {
   return lp->GetDistanceMetric(DISTANCE_TOAST_BUBBLE_ICON_SIZE);
 }
 
+views::ClientView* ToastView::CreateClientView(views::Widget* widget) {
+  views::ClientView* client_view =
+      BubbleDialogDelegateView::CreateClientView(widget);
+  client_view->SetBorder(views::CreateRoundedRectBorder(
+    kBubbleStrokeWidth, kBubbleCornerRadius, kColorToolbarSeparator));
+  return client_view;
+}
+
 void ToastView::Init() {
   ChromeLayoutProvider* lp = ChromeLayoutProvider::Get();
 
@@ -176,7 +186,7 @@ void ToastView::Init() {
 
   label_ = AddChildView(
       std::make_unique<views::Label>(toast_text_, CONTEXT_TOAST_BODY_TEXT));
-  label_->SetEnabledColor(ui::kColorToastForeground);
+  label_->SetEnabledColor(kToastForegroundColorId);
   label_->SetMultiLine(false);
   label_->SetHorizontalAlignment(gfx::ALIGN_LEFT);
   label_->SetAllowCharacterBreak(false);
@@ -200,9 +210,9 @@ void ToastView::Init() {
             base::BindRepeating(&ToastView::Close, base::Unretained(this),
                                 ToastCloseReason::kActionButton)),
         action_button_text_));
-    action_button_->SetEnabledTextColors(ui::kColorToastButton);
-    action_button_->SetBgColorIdOverride(ui::kColorToastBackgroundProminent);
-    action_button_->SetStrokeColorIdOverride(ui::kColorToastButton);
+    action_button_->SetEnabledTextColors(kToastForegroundColorId);
+    action_button_->SetBgColorIdOverride(ui::kColorSysOmniboxContainer);
+    action_button_->SetStrokeColorIdOverride(kToastForegroundColorId);
     action_button_->SetPreferredSize(gfx::Size(
         action_button_->GetPreferredSize().width(),
         lp->GetDistanceMetric(DISTANCE_TOAST_BUBBLE_HEIGHT_ACTION_BUTTON)));
@@ -228,8 +238,8 @@ void ToastView::Init() {
                                 ToastCloseReason::kCloseButton)),
         vector_icons::kCloseChromeRefreshIcon,
         lp->GetDistanceMetric(DISTANCE_TOAST_BUBBLE_ICON_SIZE),
-        ui::kColorToastForeground, ui::kColorIconDisabled,
-        ui::kColorToastForeground));
+        kToastForegroundColorId, kToastForegroundColorId,
+        kToastForegroundColorId));
     // Override the image button's border with the appropriate icon border size.
     const gfx::Insets insets =
         lp->GetInsetsMetric(views::InsetsMetric::INSETS_VECTOR_IMAGE_BUTTON);
@@ -256,8 +266,8 @@ void ToastView::Init() {
         base::RepeatingClosure(), kBrowserToolsChromeRefreshIcon,
         /*dip_size=*/
         lp->GetDistanceMetric(DISTANCE_TOAST_BUBBLE_MENU_ICON_SIZE),
-        ui::kColorToastForeground, ui::kColorIconDisabled,
-        ui::kColorToastForeground));
+        kToastForegroundColorId, kToastForegroundColorId,
+        kToastForegroundColorId));
     views::InstallCircleHighlightPathGenerator(menu_button_);
     menu_button_->SetProperty(views::kElementIdentifierKey, kToastMenuButton);
     menu_button_->SetButtonController(
@@ -316,12 +326,11 @@ void ToastView::Init() {
   }
 }
 
-void ToastView::AnimationProgressed(const gfx::Animation* animation) {
-  const double value = gfx::Tween::CalculateValue(
-      height_animation_tween_, height_animation_.GetCurrentValue());
-  const gfx::Rect current_bounds = gfx::Tween::RectValueBetween(
-      value, starting_widget_bounds_, target_widget_bounds_);
-  GetWidget()->SetBounds(current_bounds);
+void ToastView::OnBeforeBubbleWidgetInit(
+    views::Widget::InitParams* params,
+    views::Widget* widget) const {
+  BubbleDialogDelegateView::OnBeforeBubbleWidgetInit(params, widget);
+  params->shadow_type = views::Widget::InitParams::ShadowType::kNone;
 }
 
 void ToastView::AnimateIn() {
@@ -329,13 +338,6 @@ void ToastView::AnimateIn() {
     return;
   }
 
-  target_widget_bounds_ = GetWidget()->GetWindowBoundsInScreen();
-  starting_widget_bounds_ =
-      target_widget_bounds_ - gfx::Vector2d{0, kAnimationHeightOffset};
-  height_animation_tween_ = gfx::Tween::ACCEL_5_70_DECEL_90;
-  height_animation_.SetDuration(base::Milliseconds(kAnimationEntryDuration));
-  height_animation_.Start();
-
   views::View* const bubble_frame_view = GetBubbleFrameView();
   bubble_frame_view->SetPaintToLayer();
   bubble_frame_view->layer()->SetFillsBoundsOpaquely(false);
@@ -343,13 +345,13 @@ void ToastView::AnimateIn() {
       GetScaleTransformation(bubble_frame_view->bounds()));
   bubble_frame_view->layer()->SetOpacity(0);
   GetDialogClientView()->SetBackground(
-      views::CreateSolidBackground(ui::kColorToastBackgroundProminent));
+      views::CreateSolidBackground(ui::kColorSysOmniboxContainer));
   GetDialogClientView()->layer()->SetOpacity(0);
   views::AnimationBuilder()
       .Once()
       .SetDuration(base::Milliseconds(kAnimationEntryDuration))
       .SetTransform(bubble_frame_view, gfx::Transform(),
-                    height_animation_tween_)
+                    gfx::Tween::ACCEL_5_70_DECEL_90)
       .At(base::TimeDelta())
       .SetDuration(base::Milliseconds(50))
       .SetOpacity(bubble_frame_view, 1)
@@ -390,12 +392,6 @@ void ToastView::Close(ToastCloseReason r
   }
 }
 
-void ToastView::UpdateRenderToastOverWebContentsAndPaint(
-    const bool render_toast_over_web_contents) {
-  render_toast_over_web_contents_ = render_toast_over_web_contents;
-  SizeToContents();
-}
-
 gfx::Rect ToastView::GetBubbleBounds() {
   views::View* anchor_view = GetAnchorView();
   if (!anchor_view) {
@@ -405,21 +401,18 @@ gfx::Rect ToastView::GetBubbleBounds() {
   const gfx::Size preferred_size =
       GetWidget()->GetContentsView()->GetPreferredSize();
   const gfx::Rect anchor_bounds = anchor_view->GetBoundsInScreen();
+  const int window_margin = ChromeLayoutProvider::Get()->GetDistanceMetric(
+      DISTANCE_TOAST_BUBBLE_BROWSER_WINDOW_MARGIN);
 
   // A wide toast in a narrow browser window needs to be compressed to fit.
-  const int minimum_margin = ChromeLayoutProvider::Get()->GetDistanceMetric(
-                                 DISTANCE_TOAST_BUBBLE_BROWSER_WINDOW_MARGIN) -
-                             views::BubbleBorder::kShadowBlur;
   const int width =
       std::min(preferred_size.width(),
-               std::max(anchor_bounds.width() - 2 * minimum_margin, 0));
-  const int x = anchor_bounds.x() + ((anchor_bounds.width() - width) / 2);
+               std::max(anchor_bounds.width() - 2 * window_margin, 0));
 
-  // Take bubble out of its original bounds to cross "line of death", unless in
-  // fullscreen mode where the top container isn't rendered.
-  const int y = anchor_bounds.bottom() - (render_toast_over_web_contents_
-                                              ? views::BubbleBorder::kShadowBlur
-                                              : (preferred_size.height() / 2));
+  const int x = anchor_bounds.right() - width - window_margin;
+  const int y = (render_toast_over_web_contents_ ? anchor_bounds.y()
+                                                 : anchor_bounds.bottom()) +
+                window_margin;
   return gfx::Rect(x, y, width, preferred_size.height());
 }
 
@@ -431,7 +424,7 @@ void ToastView::OnThemeChanged() {
     icon_view_->SetImage(image_override_.value());
   } else {
     icon_view_->SetImage(ui::ImageModel::FromVectorIcon(
-        *icon_, color_provider->GetColor(ui::kColorToastForeground),
+        *icon_, color_provider->GetColor(kToastForegroundColorId),
         GetIconSize()));
   }
 }
@@ -450,19 +443,12 @@ void ToastView::AnimateOut(base::OnceClo
   }
 
   if (show_height_animation) {
-    starting_widget_bounds_ = GetWidget()->GetWindowBoundsInScreen();
-    target_widget_bounds_ =
-        starting_widget_bounds_ - gfx::Vector2d{0, kAnimationHeightOffset};
-    height_animation_tween_ = gfx::Tween::ACCEL_30_DECEL_20_85;
-    height_animation_.SetDuration(base::Milliseconds(kAnimationExitDuration));
-    height_animation_.Start();
-
     views::AnimationBuilder()
         .Once()
         .SetDuration(base::Milliseconds(kAnimationExitDuration))
         .SetTransform(bubble_frame_view,
                       GetScaleTransformation(bubble_frame_view->bounds()),
-                      height_animation_tween_);
+                      gfx::Tween::ACCEL_30_DECEL_20_85);
   }
 
   views::AnimationBuilder()
--- a/chrome/browser/ui/views/chrome_layout_provider.cc
+++ b/chrome/browser/ui/views/chrome_layout_provider.cc
@@ -183,31 +183,32 @@ int ChromeLayoutProvider::GetDistanceMet
     case DISTANCE_TASK_MANAGER_TAB_HEIGHT:
       return 54;
     case DISTANCE_TOAST_BUBBLE_BETWEEN_CHILD_SPACING:
-    case DISTANCE_TOAST_BUBBLE_BROWSER_WINDOW_MARGIN:
       return 4;
+    case DISTANCE_TOAST_BUBBLE_BROWSER_WINDOW_MARGIN:
+      return 6;
     case DISTANCE_TOAST_BUBBLE_BETWEEN_LABEL_ACTION_BUTTON_SPACING:
     case DISTANCE_TOAST_BUBBLE_BETWEEN_LABEL_MENU_BUTTON_SPACING:
-      return 16;
+      return 14;
     case DISTANCE_TOAST_BUBBLE_HEIGHT:
-      return 48;
+      return 32;
     case DISTANCE_TOAST_BUBBLE_HEIGHT_ACTION_BUTTON:
-      return 36;
+      return 28;
     case DISTANCE_TOAST_BUBBLE_HEIGHT_CONTENT:
       return 24;
     case DISTANCE_TOAST_BUBBLE_ICON_SIZE:
     case DISTANCE_TOAST_BUBBLE_MENU_ICON_SIZE:
-      return 20;
+      return 18;
     case DISTANCE_TOAST_BUBBLE_LEADING_ICON_SIDE_MARGINS:
       return 2;
     case DISTANCE_TOAST_BUBBLE_MARGIN_LEFT:
-      return 12;
+      return 10;
     case DISTANCE_TOAST_BUBBLE_MARGIN_RIGHT_ACTION_BUTTON:
       return 6;
     case DISTANCE_TOAST_BUBBLE_MARGIN_RIGHT_CLOSE_BUTTON:
     case DISTANCE_TOAST_BUBBLE_MARGIN_RIGHT_MENU_BUTTON:
       return 10;
     case DISTANCE_TOAST_BUBBLE_MARGIN_RIGHT_LABEL:
-      return 16;
+      return 14;
     case DISTANCE_RECENT_ACTIVITY_AVATAR_SIZE:
       return 32;
     case DISTANCE_RECENT_ACTIVITY_AVATAR_FALLBACK_SIZE:
--- a/chrome/browser/ui/toasts/toast_controller.cc
+++ b/chrome/browser/ui/toasts/toast_controller.cc
@@ -33,6 +33,9 @@
 #include "chrome/browser/ui/toasts/toast_features.h"
 #include "chrome/browser/ui/toasts/toast_metrics.h"
 #include "chrome/browser/ui/toasts/toast_view.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "chrome/browser/ui/views/frame/contents_container_view.h"
+#include "chrome/browser/ui/views/frame/contents_web_view.h"
 #include "chrome/browser/ui/views/interaction/browser_elements_views.h"
 #include "chrome/browser/ui/webui_browser/webui_browser.h"
 #include "chrome/common/pref_names.h"
@@ -46,10 +49,6 @@
 #include "ui/views/view.h"
 #include "ui/views/widget/widget.h"
 
-#if BUILDFLAG(IS_MAC)
-#include "chrome/browser/ui/fullscreen_util_mac.h"
-#endif
-
 ToastParams::ToastParams(ToastId id) : toast_id(id) {}
 ToastParams::ToastParams(ToastParams&& other) noexcept = default;
 ToastParams& ToastParams::operator=(ToastParams&& other) noexcept = default;
@@ -235,19 +234,6 @@ void ToastController::UpdateToastWidgetV
   }
 }
 
-bool ToastController::ShouldRenderToastOverWebContents() {
-  bool render_in_contents = BrowserWindowFullscreenController::From(
-                                base::to_address(browser_window_interface_))
-                                ->ShouldHideUIForFullscreen();
-
-#if BUILDFLAG(IS_MAC)
-  render_in_contents |=
-      fullscreen_utils::IsInContentFullscreen(browser_window_interface_);
-#endif
-
-  return render_in_contents;
-}
-
 void ToastController::WebContentsDestroyed() {
   omnibox_helper_observer_.Reset();
   Observe(nullptr);
@@ -292,9 +278,20 @@ void ToastController::CreateToast(ToastP
   // TODO(crbug.com/364730656): Replace this logic when improving
   // ToastController testability.
   views::View* anchor_view = nullptr;
+  bool render_toast_over_web_contents = false;
   if (browser_window_interface_) {
-    anchor_view = BrowserElementsViews::From(browser_window_interface_)
-                      ->GetView(kTopContainerElementId);
+    if (BrowserView* browser_view =
+            BrowserView::GetBrowserViewForBrowser(browser_window_interface_)) {
+      if (ContentsContainerView* contents_container_view =
+              browser_view->GetActiveContentsContainerView()) {
+        anchor_view = contents_container_view->contents_view();
+      }
+    }
+    render_toast_over_web_contents = anchor_view != nullptr;
+    if (!anchor_view) {
+      anchor_view = BrowserElementsViews::From(browser_window_interface_)
+                        ->GetView(kTopContainerElementId);
+    }
   }
   if (!anchor_view) {
     // Don't actually create the toast in unit tests
@@ -313,7 +310,7 @@ void ToastController::CreateToast(ToastP
                          params.body_string_cardinality_param);
   auto toast_view = std::make_unique<toasts::ToastView>(
       anchor_view, body_string, spec->icon(), params.image_override,
-      ShouldRenderToastOverWebContents(),
+      render_toast_over_web_contents,
       base::BindRepeating(&RecordToastDismissReason, params.toast_id));
 
   if (spec->has_close_button()) {
@@ -337,10 +334,6 @@ void ToastController::CreateToast(ToastP
   toast_view_ = toast_view.get();
   toast_widget_ =
       views::BubbleDialogDelegateView::CreateBubble(std::move(toast_view));
-  // Get rid of the border that is drawn by default when we set the toast to
-  // have a shadow.
-  toast_view_->GetBubbleFrameView()->bubble_border()->set_draw_border_stroke(
-      false);
   toast_observer_.Observe(toast_widget_);
   fullscreen_observation_.Observe(
       browser_window_interface_->GetExclusiveAccessManager()
@@ -377,8 +370,7 @@ std::u16string ToastController::FormatSt
 }
 
 void ToastController::OnFullscreenStateChanged() {
-  toast_view_->UpdateRenderToastOverWebContentsAndPaint(
-      ShouldRenderToastOverWebContents());
+  toast_view_->SizeToContents();
 }
 
 void ToastController::ClearTabScopedToasts() {
--- a/chrome/browser/ui/toasts/toast_view.h
+++ b/chrome/browser/ui/toasts/toast_view.h
@@ -21,6 +21,7 @@ class MenuModel;
 }  // namespace ui
 
 namespace views {
+class ClientView;
 class ImageButton;
 class ImageView;
 class Label;
@@ -47,8 +48,7 @@ enum class ToastCloseReason {
 // LINT.ThenChange(//tools/metrics/histograms/metadata/toasts/enums.xml:ToastCloseReason)
 
 // The view for toasts.
-class ToastView : public views::BubbleDialogDelegateView,
-                  public views::AnimationDelegateViews {
+class ToastView : public views::BubbleDialogDelegateView {
   METADATA_HEADER(ToastView, views::BubbleDialogDelegateView)
 
  public:
@@ -61,7 +61,7 @@ class ToastView : public views::BubbleDi
       const std::u16string& toast_text,
       const gfx::VectorIcon& icon,
       std::optional<ui::ImageModel> image_override,
-      bool should_hide_ui_for_fullscreen,
+      bool render_toast_over_web_contents,
       base::RepeatingCallback<void(ToastCloseReason)> on_toast_close_callback);
   ~ToastView() override;
 
@@ -81,20 +81,19 @@ class ToastView : public views::BubbleDi
   // views::BubbleDialogDelegateView::CreateBubble).
   void AddMenu(std::unique_ptr<ui::MenuModel> model);
 
+  // views::BubbleDialogDelegate:
+  views::ClientView* CreateClientView(views::Widget* widget) override;
+
   // views::BubbleDialogDelegateView:
   void Init() override;
-
-  // AnimationDelegateViews:
-  void AnimationProgressed(const gfx::Animation* animation) override;
+  void OnBeforeBubbleWidgetInit(views::Widget::InitParams* params,
+                                views::Widget* widget) const override;
 
   void AnimateIn();
 
   // Animates out the toast, then closes the toast widget.
   void Close(ToastCloseReason close_reason);
 
-  void UpdateRenderToastOverWebContentsAndPaint(
-      const bool render_toast_over_web_contents);
-
   views::Label* label_for_testing() { return label_; }
   views::MdTextButton* action_button_for_testing() { return action_button_; }
   views::ImageButton* close_button_for_testing() { return close_button_; }
@@ -117,15 +116,10 @@ class ToastView : public views::BubbleDi
   // Resets `menu_runner_` and removes the highlight from the menu button.
   void OnMenuClosed();
 
-  gfx::LinearAnimation height_animation_{this};
-  gfx::Rect starting_widget_bounds_;
-  gfx::Rect target_widget_bounds_;
-  gfx::Tween::Type height_animation_tween_;
-
   const std::u16string toast_text_;
   const raw_ref<const gfx::VectorIcon> icon_;
   const std::optional<ui::ImageModel> image_override_;
-  bool render_toast_over_web_contents_;
+  const bool render_toast_over_web_contents_;
   bool has_close_button_ = false;
   bool has_action_button_ = false;
   std::u16string action_button_text_;
