--- a/net/dns/public/doh_provider_entry.cc
+++ b/net/dns/public/doh_provider_entry.cc
@@ -314,7 +314,7 @@ const DohProviderEntry::List& DohProvide
        {
            "Quad9Secure",
            MAKE_STATIC_STORAGE_BASE_FEATURE(kDohProviderQuad9Secure,
-                                            base::FEATURE_DISABLED_BY_DEFAULT),
+                                            base::FEATURE_ENABLED_BY_DEFAULT),
            {"9.9.9.9", "149.112.112.112", "2620:fe::fe", "2620:fe::9"},
            /*dns_over_tls_hostnames=*/{"dns.quad9.net", "dns9.quad9.net"},
            "https://dns.quad9.net/dns-query",
